cdk/domino_cdk/config/base.py

```diff
@@ -102,6 +102,11 @@ def val(path: str, obj):
                         [val(f"{path}.{f.name}.{k}", v) for k, v in value.items() if is_dataclass(v)]
                 elif value and f.type != type(value):
                     errors.append(f"{path}.{f.name} type ({f.type}) does not match value: [{value}] ({type(value)})")
+                if f.name == "tags":
+                    if "domino-deploy-id" in value:
+                        errors.append(f"Tag domino-deploy-id cannot be overridden")
+                    else:
+                        value["domino-deploy-id"] = self.name
```

Comment:
Doing this here has a round-trip problem, since we render the config template based on a loaded, running config (ie where the value is already injected). Both template generation and schema upgrades would both cause an impact here. I think it would be reflected in the unit tests, but this branch predates them?

If we're doing it via the config, I think this value should be set in the config template, and then if there's a check here it should be that the value is what we expect it to be.

Alternate version of the "get tags function" : If we make a function that's _part of the config object_ to get the tags, I think that's pretty clean?

So then we can call something like `self.cfg.get_tags()`, and it'd return the tags + domino-deploy-id tag, but it's still never stored in the tags variable of the config object so doesn't run into this issue?

Reply:
ok.